lib/matplotlib/axes/_axes.py

```diff
@@ -5640,7 +5641,9 @@ def hist(self, x, bins=10, range=None, normed=False, weights=None,
 
         # We need to do to 'weights' what was done to 'x'
         if weights is not None:
-            if isinstance(weights, np.ndarray) or not iterable(weights[0]):
+            if input_empty:
+                w = np.array([])
```

Comment:
This seems wrong to me as it is discarding the user input in the case of empty weights.  I don't think this block of changes is required